pynicotine/transfers.py

```diff
@@ -455,22 +455,26 @@ def slot_limit_reached(self):
             maxupslots = 1
 
         in_progress_count = 0
-        now = time.time()
+        current_time = time.time()
 
-        for i in self.uploads:
+        for i in reversed(self.uploads):
             if i.conn is not None and i.speed is not None:
                 # Currently transferring
                 in_progress_count += 1
 
-            elif (now - i.last_status_change) < 30:
+            elif (current_time - i.last_status_change) < 30:  # ToDo: performance bottleneck
                 # Transfer initiating, changed within last 30 seconds
```

Comment:
When there are over 2000 historic uploads to iterate, testing this condition can become very resource intensive when there are many queued transfers and a low number of slots is set. Breaking out of the loop seem to work around this, but maybe it is redundant to check this arbitrary state like this, maybe the next conditions are faster so they can be processed first?